### PR TITLE
[Storage] Bucket Policy Only samples

### PR DESCRIPTION
**Pull request by frankyn**
Samples for the GCS Bucket Policy Only feature. Includes some reformatting.

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java [hunk 1]
```diff
@@ -1393,4 +1526,74 @@ public Blob releaseTemporaryHold(String bucketName, String blobName) throws Stor
...
+  /**
+   * Example of how to enable Bucket Policy Only for a bucket
+   */
+  public Bucket enableBucketPolicyOnly(String bucketName) throws StorageException {
+    // [START storage_enable_bucket_policy_only]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
```
Comment (kolea2) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: I think this should be uncommented?

Comment (kolea2) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: Oh nvm, I see it's being passed through as a parameter. Can we remove the passed in parameter and leave the code above for clarity?

Comment (frankyn) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: @kolea2, thanks for the feedback.

The passed in parameter is used by the integration test and the commented variable could be copy/pasted by a developer into their code. I'm using existing sample design followed by this file. 

Could you add an example of what you'd like to see?

Comment (kolea2) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: Ah, ok makes sense - didn't realize the parameters were used in that way. That seems like the best approach then!

**Comment by frankyn** [comment]
Thanks @kolea2! I'm running kokoro which didn't run on this PR yet. Is that a bug?

**Pull request merged**